### PR TITLE
Move batch job setup to DockerBatchBase and add a unit test

### PR DESCRIPTION
**Pull request by nweires**
Move the cloud-agnostic logic for setting up batches of simulations into the DockerBatchBase class. This function will be shared by the AWS and GCP implementations.

Also add a unit test for this function.

## Checklist

Not all may apply

- [x] Code changes (must work)
- [x] Tests exercising your feature/bug fix (check coverage report on Checks -> BuildStockBatch Tests -> Artifacts)
- [x] Coverage has increased or at least not decreased. Update `minimum_coverage` in `.github/workflows/ci.yml` as necessary.
- [x] All other unit and integration tests passing
- [x] ~~Update validation for project config yaml file changes~~
- [x] ~~Update existing documentation~~
- [x] ~~Run a small batch run on Eagle to make sure it all works if you made changes that will affect Eagle~~
  - Only touches AWS implementation
- [x] ~~Add to the changelog_dev.rst file and propose migration text in the pull request~~
  - No behavior changes - just refactoring

### CODE REVIEW

File: buildstockbatch/cloud/docker_base.py [hunk 1]
```diff
@@ -43,3 +55,131 @@ def docker_image(self):
...
+        """
+        Prepare batches of samples to be uploaded and run in the cloud.
+
+        This will:
+            - Perform the sampling
+            - Split the samples into (at most) self.batch_array_size batches
+            - Collect and package and the required assets, including weather
+              files, and write them to tmppath.
+
+        self.weather_dir must exist before calling this method. This is where weather files are stored temporarily.
```
Comment (lathanh) [inline buildstockbatch/cloud/docker_base.py:69]: Does this need to be a property? It doesn't seem to be used by anything else and would be simpler and more straightforward to just pass in as a parameter.

**Review by lathanh — APPROVED** [review]

File: buildstockbatch/cloud/docker_base.py (hunk 1 above)
Comment (nweires) [reply buildstockbatch/cloud/docker_base.py:69]: I thought so too, but it turns out it's used in the base class [here](https://github.com/rewiringamerica/buildstockbatch/blob/661f66abc1e78faf47c9ba56c07ba0918278a15c/buildstockbatch/base.py#L102).

Comment (lathanh) [reply buildstockbatch/cloud/docker_base.py:69]: ah, ok :grimacing:

**Review by nmerket — CHANGES_REQUESTED** [review]
For whatever reason my pull request template stopped working a while ago. I think I fixed it now. Either way, I pasted into the description on this PR the checklist we ask people to run through for each PR. Some of them may not apply in which case, ~~cross them out~~ and check the box. This helps me to ensure all the steps are complete.

**Review by nmerket — APPROVED** [review]
Seems pretty straightforward to me. Thanks for adding a unit test.

**Pull request merged**